packages/compass-e2e-tests/helpers/commands/connect-form.ts

```diff
@@ -0,0 +1,1013 @@
+  const element = await browser.$(selector);
+  if (!(await element.isExisting())) {
+    return null; // as opposed to true for checked and false for not
+  }
+
+  return element.isSelected();
+}
+
+async function getText(
+  browser: CompassBrowser,
+  selector: string
+): Promise<string | null> {
+  const element = await browser.$(selector);
+  if (!(await element.isExisting())) {
+    return null;
+  }
+
+  const text = await element.getText();
+  return text || null;
+}
+
+async function getFilename(
+  browser: CompassBrowser,
+  selector: string
+): Promise<string | null> {
+  const text = await getText(browser, selector);
+  return text === 'Select a file...' ? null : text;
+}
+
+async function getValue(
+  browser: CompassBrowser,
+  selector: string
+): Promise<string | null> {
+  const element = await browser.$(selector);
+  if (!(await element.isExisting())) {
+    return null;
+  }
+
+  const value = await element.getValue();
+  return value || null;
+}
+
+async function getMultipleValues(
+  browser: CompassBrowser,
+  selector: string
+): Promise<string[] | null> {
+  const results = (
+    await browser.$$(selector).map((element) => element.getValue())
+  ).filter((result) => result !== '');
+
+  return results.length ? results : null;
+}
+
+interface NamedPromises {
+  [key: string]: Promise<any>;
+}
+
+async function promiseMap(map: NamedPromises) {
+  const results = await Promise.all(Object.values(map));
+  return Object.fromEntries(
+    Object.keys(map)
+      .map((k, i) => [k, results[i]])
+      .filter(([, v]) => v !== null)
+  );
+}
+
+async function waitForElementAnimations(browser: CompassBrowser, element: any) {
+  let previousResult = {
+    ...(await element.getLocation()),
+    ...(await element.getSize()),
+  };
+  await browser.waitUntil(async function () {
+    // small delay to make sure that if it is busy animating it had time to move
+    // before the first check and between each two checks
+    await browser.pause(50);
+
+    const result = {
+      ...(await element.getLocation()),
+      ...(await element.getSize()),
+    };
+    const stopped = _.isEqual(result, previousResult);
+    previousResult = result;
+    return stopped;
+  });
+}
+
+const colorMap: Record<string, string> = {
+  'no-color': 'No Color',
+  color1: 'Red',
+  color2: 'Pink',
+  color3: 'Orange',
+  color4: 'Yellow',
+  color5: 'Green',
+  color6: 'Teal',
+  color7: 'Blue',
+  color8: 'Iris',
+  color9: 'Purple',
+};
+
+function colorValueToName(color: string): string {
+  if (colorMap[color]) {
+    return colorMap[color];
+  }
+  return color;
+}
+
+export async function setConnectFormState(
+  browser: CompassBrowser,
+  state: ConnectFormState
+): Promise<void> {
+  await browser.resetConnectForm();
+
+  // Something to keep in mind is that if you specify both connectionString AND
+  // other options, then the other options are going to override the
+  // connectionString. You probably want just one or the other.
```

Comment:
is it a hard rule that could even be validated against and throw, or more of a rule of thumb?

Reply:
Rule of thumb, I think.